### PR TITLE
Implementation of feature request 54

### PR DESCRIPTION
**Pull request by rgarcia89**
### Fixes: Feature Request #54 

Implementes what I asked for in #54 
I will add more stuff to it within the coming weeks like email notification for peers that are down for longer than 3 days same as auto disabling for peers that are longer down than 2 weeks.

### CODE REVIEW

**Comment by jlixfeld** [comment]
Hi,

I've opened #53 in hopes to extend the info presented about why a session is down.  Wondering if you have any thoughts on how to do that.  It might fit in with what you're working on here?

Also, would you mind opening issues for your email notification ideas?  I have some thoughts on that which I'd like to share.

File: peering/models.py [hunk 1]
```diff
@@ -645,6 +646,9 @@ def update_peering_session_states(self):
                                 ip_address, self.name.lower(), state)
 
                             # Update the BGP state of the session
+                            if (peering_session.bgp_state == "established"):
```
Comment (respawner) [inline peering/models.py]:
```suggestion
                            if (peering_session.bgp_state == BGP_STATE_ESTABLISHED):
```

File: peering/models.py [hunk 2]
```diff
@@ -645,6 +646,9 @@ def update_peering_session_states(self):
...
+                                peering_session.last_state_established = timezone.localdate()
```
Comment (respawner) [inline peering/models.py]:
```suggestion
                                peering_session.last_state_established = timezone.now()
```

File: peering/models.py [hunk 3]
```diff
@@ -205,6 +205,7 @@ class BGPSession(CreatedUpdatedModel):
     received_prefix_count = models.PositiveIntegerField(blank=True, null=True)
     advertised_prefix_count = models.PositiveIntegerField(blank=True,
                                                           null=True)
+    last_state_established = models.DateField(blank=True, null=True)
```
Comment (respawner) [inline peering/models.py]:
```suggestion
    last_established_state = models.DateField(blank=True, null=True)
```

File: templates/peering/session/details.html [hunk 4]
```diff
@@ -51,6 +51,10 @@ <h1>{% block title %}{{ peering_session }}{% endblock %}</h1>
                 <td>IP Address</td>
                 <td>{{ peering_session.ip_address }}</td>
               </tr>
+              <tr>
+                <td>Last Established</td>
+                <td>{{ peering_session.last_state_established }}</td>
```
Comment (respawner) [inline templates/peering/session/details.html]:
```suggestion
                <td>{{ peering_session.last_established_state }}</td>
```

File: templates/peering/session/direct/details.html [hunk 5]
```diff
@@ -57,6 +57,10 @@ <h1>{% block title %}{{ peering_session }}{% endblock %}</h1>
                 <td>IP Address</td>
                 <td>{{ peering_session.ip_address }}</td>
               </tr>
+              <tr>
+                <td>Last Established</td>
+                <td>{{ peering_session.last_state_established }}</td>
```
Comment (respawner) [inline templates/peering/session/direct/details.html]:
```suggestion
                <td>{{ peering_session.last_established_state }}</td>
```

**Pull request merged**